### PR TITLE
Add services parent to subsite overviews #148

### LINKED ISSUE #148: Subsite overview page needs a parent field
In https://github.com/localgovdrupal/localgov/issues/487 we made parent field location and labelling consistent 

We need to ensure the same with Subsite overview pages. They should have a parent field:

- At the top of the description tab
- Labelled 'parent page'

This is quite urgent as two new councils (Essex and Dumfries) are starting to use subsites heavily this month

### PR DESCRIPTION
**Pull request by stephen-cox**
Adds a parent page field to subsite overview pages. When this is used the subsite will appear in the services heirachy.

image

Fixes #148

### CODE REVIEW

**Comment by willguv** [comment]
The screenshot looks fine @stephen-cox - thanks for working on it

Don't think I've got the power to merge this PR though :)

**Comment by stephen-cox** [comment]
> Don't think I've got the power to merge this PR though :)

Thanks for checking @willguv - we'll merge and release later at Merge Tuesday

**Comment by finnlewis** [comment]
Discussing in Merge Tuesday: 
 - Ekes points out that we have other fields that could be parent pages, and we might want to mention that this is a services parent page. For example, if posting into directories a directory might be a parent. 
 - Ru concurs. Working in subsites, people refer to parent pages to mean the subsite overview, but also the services parent page.

Q1: Should we name this "Service parent page" ?

Q2. What is the expected behaviour of setting this field? 
 - By selecting a parent service landing page or service sub-landing page:
 -  - 1. it sets the url alias to be child of the parent, which in turn sets the breadcrumb.
 -  - 2. It then gets listed as a potential child in service landing pages. 
 
Also noting that this is just for new installs for now, so do we need documentation on how to implement on existing sites? 

See https://github.com/localgovdrupal/localgov_guides/pull/166

I think I need to read up on th original issue https://github.com/localgovdrupal/localgov/issues/487 

I'll do that this week.

Note: everyone is in agreement that consistency is important, just co-ordinating on the various parent fields on names is key.

**Comment by willguv** [comment]
Thanks for these questions @finnlewis 

Q1

I discussed naming of parent fields with content designers when we were rationalising them a while ago. "Parent page" was decided upon. 

Presumably a content designer would only set a parent page that would make sense, and we only allow parent fields to be set that make sense. So does this field name need to be anything other than "Parent page"? 

Sorry if I'm missing something here.

Q2

The expected behavior is as you describe. It's valid that subsite overview pages are listed in service landing pages as potential children, and I've seen content designers try to do this.

Is there a risk in adding this to all new sites, not just new installs? Most councils are likely to use it and can ignore if not. If there's a risk, agree a doc is the right way to go.

File: localgov_subsites.install [hunk 1]
```diff
@@ -5,6 +5,16 @@
  * LocalGov Subsites install file.
  */
 
+/**
+ * Implements hook_install().
+ */
+function localgov_subsites_install() {
```
Comment (ekes) [inline localgov_subsites.install]: As we're going to be altering config here I think we want a `$is_syncing` here and check that it's not before updating the config.

File: config/install/field.field.node.localgov_subsites_page.localgov_subsites_parent.yml [hunk 2]
```diff
@@ -11,7 +11,7 @@ id: node.localgov_subsites_page.localgov_subsites_parent
 field_name: localgov_subsites_parent
 entity_type: node
 bundle: localgov_subsites_page
-label: Parent
+label: 'Subsite parent page'
```
Comment (ekes) [inline config/install/field.field.node.localgov_subsites_page.localgov_subsites_parent.yml:14]: :smile:

**Review by ekes — CHANGES_REQUESTED** [review]

File: localgov_subsites.install (hunk 1 above)
Comment (stephen-cox) [reply localgov_subsites.install]: Indeed it should. Just added a check for this.

**Review by ekes — APPROVED** [review]

**Pull request merged**